text/0225-configuration.md

```diff
@@ -0,0 +1,411 @@
+  #
+  # Environment variable: OTEL_PROPAGATORS
+  propagators: [tracecontext, baggage]
+  # Configure the tracer provider.
+  tracer_provider:
+    # Span exporters. Each exporter key refers to the type of the exporter. Values configure the exporter. Exporters must be associated with a span processor.
+    exporters:
+      # Configure the zipkin exporter.
+      zipkin:
+        # Sets the endpoint.
+        #
+        # Environment variable: OTEL_EXPORTER_ZIPKIN_ENDPOINT
+        endpoint: http://localhost:9411/api/v2/spans
+        # Sets the max time to wait for each export.
+        #
+        # Environment variable: OTEL_EXPORTER_ZIPKIN_TIMEOUT
+        timeout: 10000
+    # List of span processors. Each span processor has a name and args used to configure it.
+    span_processors:
+      # Add a batch span processor.
+      #
+      # Environment variable: OTEL_BSP_*, OTEL_TRACES_EXPORTER
+      - name: batch
+        # Configure the batch span processor.
+        args:
+          # Sets the delay interval between two consecutive exports.
+          #
+          # Environment variable: OTEL_BSP_SCHEDULE_DELAY
+          schedule_delay: 5000
+          # Sets the maximum allowed time to export data.
+          #
+          # Environment variable: OTEL_BSP_EXPORT_TIMEOUT
+          export_timeout: 30000
+          # Sets the maximum queue size.
+          #
+          # Environment variable: OTEL_BSP_MAX_QUEUE_SIZE
+          max_queue_size: 2048
+          # Sets the maximum batch size.
+          #
+          # Environment variable: OTEL_BSP_MAX_EXPORT_BATCH_SIZE
+          max_export_batch_size: 512
+          # Sets the exporter. Exporter must refer to a key in sdk.tracer_provider.exporters.
+          #
+          # Environment variable: OTEL_TRACES_EXPORTER
+          exporter: zipkin
+  # Configure the meter provider.
+  ...
+```
+
+### Environment variable substitution
+
+Configuration files *MUST* support environment variable expansion. While this accommodates the scenario in which a configuration file needs to reference sensitive data and is not able to be stored securely, environment variable expansion is not limited to sensitive data.
+
+The syntax for environment variable expansion mirrors the [collector](https://opentelemetry.io/docs/collector/configuration/#configuration-environment-variables).
+
+For example, given an environment where `API_KEY=1234`, the configuration file contents:
+
+```yaml
+scheme_version: 0.1
+sdk:
+  tracer_provider:
+    exporters:
+      otlp:
+        endpoint: https://example.host:4317/v1/traces
+        headers:
+          api-key: ${env:API_KEY}
+```
+
+Result in the following after substitution:
+
+```yaml
+scheme_version: 0.1
+sdk:
+  tracer_provider:
+    exporters:
+      otlp:
+        endpoint: https://example.host:4317/v1/traces
+        headers:
+          api-key: 1234
```

Comment:
For the config properties with a certain format expected, such as headers, should give the percent encoded value as input? or Will configuration object take care of that detail?

Reply:
That's a great question. For the properties we have specific requirements about, the schema should validate for the expected types.

Reply:
The [OTEL_EXPORTER_OTLP_HEADERS](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#specifying-headers-via-environment-variables) accepts plain text (presumably since it doesn't say anything about percent encoding) so I think the same should be true here.